lib/logstash/plugin_mixins/rabbitmq_connection.rb

```diff
@@ -96,9 +93,12 @@ def rabbitmq_settings
 
         s[:timeout] = @connection_timeout || 0
         s[:heartbeat] = @heartbeat || 0
-        s[:tls] = @ssl if @ssl
-        s[:tls_certificate_path] = @tls_certificate_path || ""
-        s[:tls_certificate_password] = @tls_certificate_password || ""
+
+        # This MUST be passed as a string or the cert options will be ignored
+        s[:tls] = @ssl.is_a?(String) ? @ssl : "TLSv1.2"
```

Comment:
Actually, the problem with specifying TLSv1.2 as the default is that it would turn TLS on by default. I don't think we want to make that the case.

Thinking through the :validate block, I'm worried that we'll then have to maintain this if TLSv1.3 becomes available.

In the documentation I only give "TLSv1.2" as an example since its really the best choice.

Thoughts?